GhostNetwork.Profiles.Domain/SecuritySettings/ISecuritySettingService.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Domain;
+
+namespace GhostNetwork.Profiles.SecuritySettings
+{
+    public interface ISecuritySettingService
+    {
+        Task<DomainResult> UpdateAsync(Guid profileId, Access accessForPosts, Guid[] certainUsersForPosts, Access accessForFriends, Guid[] certainForFriends);
+
+        Task<(DomainResult, SecuritySetting)> GetByUserIdAsync(Guid profileId);
+    }
+
+    public class SecuritySettingsService : ISecuritySettingService
+    {
+        private readonly ISecuritySettingStorage securitySettingsStorage;
+        private readonly IProfileStorage profileStorage;
+
+        public SecuritySettingsService(ISecuritySettingStorage securitySettingsStorage, IProfileStorage profileStorage)
+        {
+            this.securitySettingsStorage = securitySettingsStorage;
+            this.profileStorage = profileStorage;
+        }
+
+        public async Task<DomainResult> UpdateAsync(Guid profileId, Access accessForPosts, Guid[] certainUsersForPosts, Access accessForFriends, Guid[] certainUsersForFriends)
+        {
+            var profile = await profileStorage.FindByIdAsync(profileId);
+            if (profile == null)
+            {
+                return DomainResult.Error("Profile not found");
+            }
+
+            var security = await securitySettingsStorage.FindByUserIdAsync(profileId);
+            if (security == null)
+            {
+                security = new SecuritySetting(
+                    default,
+                    profileId,
+                    accessForPosts,
+                    accessForFriends,
+                    certainUsersForPosts.ToList(),
+                    certainUsersForFriends.ToList());
+            }
+            else
+            {
+                security.Update(accessForPosts, accessForFriends, certainUsersForPosts.ToList(), certainUsersForFriends.ToList());
+            }
+
+            await securitySettingsStorage.UpdateAsync(profileId, security);
+            return DomainResult.Success();
+        }
+
+        public async Task<(DomainResult, SecuritySetting)> GetByUserIdAsync(Guid profileId)
```

Comment:
As a customer of this API I do not expect to get error. Let’s return null in this case (404 status). Also remove DomainResult from return type